packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -169,15 +176,30 @@ function buildNumericRanges(ranges: string[]) {
     .map(([start, end]) => buildNumericRange({start, end, state: 'selected'}));
 }
 
+function isValidDateRangeValue(date: string) {
+  try {
+    if (isSearchApiDate(date)) {
+      validateAbsoluteDate(date, API_DATE_FORMAT);
+      return true;
+    }
+    if (isRelativeDateFormat(date)) {
+      validateRelativeDate(date);
+      return true;
+    }
+
+    return false;
+  } catch (error) {
+    return false;
+  }
+}
+
 function buildDateRanges(ranges: string[]) {
   return ranges
     .map((str) => str.split(rangeDelimiter))
     .filter(
       (range) =>
         range.length === 2 &&
-        range.every(
-          (value) => isSearchApiDate(value) || isRelativeDateFormat(value)
-        )
+        range.every((value) => isValidDateRangeValue(value))
```

Comment:
That's an entrypoint for the Url manager, if a customer enters an invalid value, it won't be added to the request and will be filtered out of the parameters